pilot/pkg/model/gateway_test.go

```diff
@@ -56,6 +57,14 @@ func TestMergeGateways(t *testing.T) {
 			map[string]int{"http.7": 2},
 			2,
 		},
+		{
+			"two servers on the same port with different bind",
+			[]config.Config{gwHTTPbar, gwHTTPlocalbar},
+			2,
+			2,
+			map[string]int{"http.7": 2},
```

Comment:
This is not right - we need different routes for this one. Otherwise, a user will create a 127.0.0.1 gateway, bind a super secret VirtualService to it. Then, another user will create a 0.0.0.0 Gateway and suddenly the secret VS is exposed externally?

If I am understanding right.

Reply:
Yup. I realized that. Thanks for catching it. Will fix it.

Reply:
I handled this case. PTAL